zip-0226.rst

```diff
@@ -1,8 +1,313 @@
+
+Specification
+=============
+
+Most of the protocol is kept the same as the Orchard protocol released with NU5, except for the following.
+
+Asset Identifiers
+-----------------
+
+For every new Asset, there must be a new and unique identifier of the Asset. Every Asset is defined by the *Asset description*, :math:`\mathsf{asset\_desc}`, which is a global byte string (valid across all -future- Zcash protocols). From this Asset description the specific Asset identifier, :math:`\mathsf{AssetId}`, is derived with a `GroupHash` and used within the output notes, as defined by the specific protocol (e.g.: for now Orchard-style, using Pallas curve).
+
+This :math:`\mathsf{AssetId}` will be the base point of the value commitment for the specific custom Asset. Note that the :math:`\mathsf{AssetId}` of the ZEC Asset will be kept as the original value base point, :math:`\mathcal{V}^\mathsf{Orchard}`.
+
+In future network and protocol upgrades, the same Asset description string can be carried on, with potentially a mapping into a different shielded protocol. In that case, the turnstile should know how to transform the Asset identifier, :math:`\mathsf{AssetId}`, from one shielded protocol to another.
+
+Note Structure & Commitment
+---------------------------
+
+First, we need to adapt the components that define the Assets, i.e.: *notes*. A ZSA note differs from an Orchard note [#protocol-notes]_ by additionally including the identifier of the Asset, :math:`\mathsf{AssetId}`. So a ZSA note is a tuple :math:`(\mathsf{g_d, pk_d, v, \rho, \psi, AssetId})`,
+where 
+
+- :math:`\mathsf{AssetId} : \mathbb{P}` is the unique random element of the Pallas group [#protocol-pallasandvesta]_ that identifies each Asset in the Orchard protocol. 
+
+Let :math:`\mathsf{Note^{OrchardZSA}}` be the type of a ZSA note, i.e. 
+:math:`\mathsf{Note^{OrchardZSA}} := \mathsf{Note^{Orchard}} \times \mathbb{P}`.
+
+In this case, the instance of the note commitment scheme, :math:`\mathsf{NoteCommit^{OrchardZSA}_{rcm}}`, will differ from :math:`\mathsf{NoteCommit^{Orchard}_{rcm}}` in that for non-ZEC Assets, the Asset identifier will be added as an input to the commitment computation. As we will see, the nested structure of the Sinsemilla-based commitment [#protocol-concretesinsemillacommit]_ allows us to add the Asset identifier as a final recursive step, and hence keep a single instance of the Sinsemilla hash function in the circuit for the note commitment verification.
+
+The commitment output is still indistinguishable from the original Orchard ZEC note commitments, by definition of the Sinsemilla hash function [#protocol-concretesinsemillahash]_. ZSA note commitments will therefore be added to the same Orchard Note Commitment Tree. In essence, we have:
+
+.. math:: \mathsf{NoteCommit^{OrchardZSA}_{rcm}(repr_{\mathbb{P}}(g_d), repr_{\mathbb{P}}(pk_d), v, \rho, \psi, AssetId)} \in \mathsf{NoteCommit^{Orchard}.Output}
```

Comment:
This describes the rationale for the approach taken, but is not a specification of the approach. Rework this to specify how the note commitment for an Orchard ZSA note is computed, optionally including the rationale afterwards (see [ZIP 317](https://zips.z.cash/zip-0317#rationale-for-block-template-construction-algorithm) for an example of doing this in-line):

```suggestion
The note commitment for :math:`\mathsf{Note^{OrchardZSA}}` is:

 :math:`\mathsf{NoteCommit^{OrchardZSA}_{rcm}} = ...`

 
Rationale for Orchard ZSA note commitment
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The nested structure of the Sinsemilla-based commitment [#protocol-concretesinsemillacommit]_ allows us to add the Asset identifier as a final recursive step, and hence keep a single instance of the Sinsemilla hash function in the circuit for the note commitment verification.

The commitment output is still indistinguishable from the original Orchard ZEC note commitments, by definition of the Sinsemilla hash function [#protocol-concretesinsemillahash]_. ZSA note commitments will therefore be added to the same Orchard Note Commitment Tree. In essence, we have:

.. math:: \mathsf{NoteCommit^{OrchardZSA}_{rcm}(repr_{\mathbb{P}}(g_d), repr_{\mathbb{P}}(pk_d), v, \rho, \psi, AssetId)} \in \mathsf{NoteCommit^{Orchard}.Output}
```

Alternatively, move the rationale to the bottom of the document in a common "Rationale" section (see [ZIP 244](https://zips.z.cash/zip-0244#rationale) for an example of this).

In general, the intention here is that the "Specification" section should be sufficiently clear and standalone that it can be correctly implemented without any of the rationale.